README.md

```diff
@@ -37,7 +37,7 @@ For Ginkgo core library:
 
 *   _cmake 3.9+_
 *   C++11 compliant compiler, one of:
-    *   _gcc 5.3+, 6.3+, 7.3+, 8.1+_
+    *   _gcc 5.3+, 6.3+, 7.3+, 8.1+, 9.1+, 10.1+_
```

Comment:
Maybe we should also list the clang versions we support, and remove the __TODO__ from AppleClang?

Reply:
For clang, I don't think there really is any difference between the major releases in our case, since 3.9 and full C++11 support there has been no major problem. For GCC, the issue was that several major versions exist next to each other, and C++ 11 support was incomplete for example until 5.3 (at least, there were bugs with the compiler). I remember I did the work of fixing the issues we had with 5.3 some time ago, but going further down was too much work.

For AppleClang, as I have no MacOS available I cannot judge.

Reply:
I see your point about clang. My main hold-up is that since we list versions explicitly for GCC, readers might assume that we only support a very old clang, but that is only a very vague issue.

We could list AppleClang 11, since that's what the Github Action uses.

Reply:
I changed this to say `all versions after 8.1+`. I think this shows that the previous listings are an exception and otherwise we support all versions after the ones listed here.